### ISSUE TITLE
Remove the warning note about skipping, and instead show the warning when selecting skip

### ISSUE THREAD

**Issue by christianrondeau** [chocolatey/choco#183]
When asking to run a PowerShell install script, the prompt looks like this:

```bash
Do you want to run the script?
 NOTE: If you choose not to run the script, the installation will
 fail.
 Skip is an advanced option and most likely will never be wanted.

 1) yes
 2) no [Default - Press Enter]
 3) skip
```

I suggest we do *not* show that NOTE, and instead ask for another confirmation when selecting skip (note that I'm using the #689 representation):

```bash
Do you want to run the script? (Yes/No/Skip): s
NOTE: The installation will fail. Skip is an advanced option and most likely will never be wanted.
Skip anyway? (Yes/No):
```

The message could probably also be improved a little bit.

For more details: https://groups.google.com/forum/#!topic/chocolatey/UVncL7PxXRg

(Moved from https://github.com/chocolatey/chocolatey/issues/691)

**Comment by ferventcoder** [comment]
:+1:

**Comment by christianrondeau** [comment]
So, let's see if all flows make sense. My initial suggestion in the original post means that if you skip and then press no, do you go back to the yes/no/skip?

Also, if I understand correctly, selecting no will *fail* the installation, rather than just *cancel* it? Otherwise I would change the messages.

Here is an example of what the flow could look like:

```bash
Do you want to run the script? (Yes/No/Skip): s
NOTE: Skip is an advanced option and most likely will never be wanted.
Skip anyway? (Yes/No): n
Do you want to run the script? (Yes/No/Skip): n
```

Note that I get rid of the "installation will fail", if my understanding that no == cancel is correct.

**Comment by ferventcoder** [comment]
No means install will fail.

On Wednesday, March 25, 2015, Christian Rondeau <notifications@github.com>
wrote:

> So, let's see if all flows make sense. My initial suggestion in the
> original post means that if you skip and then press no, do you go back to
> the yes/no/skip?
>
> Also, if I understand correctly, selecting no will *fail* the
> installation, rather than just *cancel* it? Otherwise I would change the
> messages.
>
> Here is an example of what the flow could look like:
>
> Do you want to run the script? (Yes/No/Skip): s
> NOTE: Skip is an advanced option and most likely will never be wanted.
> Skip anyway? (Yes/No): n
> Do you want to run the script? (Yes/No/Skip): n
>
> Note that I get rid of the "installation will fail", if my understanding
> that no == cancel is correct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chocolatey/choco/issues/183#issuecomment-85975637>.
>


-- 

Rob
"Be passionate in all you do"

http://devlicio.us/blogs/rob_reynolds
http://ferventcoder.com
http://twitter.com/ferventcoder

**Comment by christianrondeau** [comment]
So, what is the difference between skip and install then? The only difference I see between both is in `PowerShellServices.run_action`, where there is an ExitCode set to -1... I didn't find what else it changed to the flow.

I believe this warrants better explanations (as a user, I am also confused), or a change so that you can either skip the package (no), or install it (yes) but no middle-ground.

Also, let me know if you like the workflow suggested in my last comment!

**Comment by ferventcoder** [comment]
Skip means install without running powershell. It may be better just to move it back to the option you pass during install and not something we ask. So I think `(yes/no/view)` - view is a different pr, no?

**Comment by ferventcoder** [comment]
> `Do you want to run the script? (Yes/No/Skip): n`

This might be best handled as `Do you want to run the script? ([y]es/[n]o): n`

**Comment by christianrondeau** [comment]
Note 1: About your last comment, this would be part of #181. Note that I did that first, then looked up other console tools. Most of them just ask y/n or yes/no. The console will accept both `y` and `yes`, so I guess the visual cue is unnecessary. Let's move the discussion back in #181 for this, since the commit is already done and ready!

Note 2: For the option to optionally "view" the script, yes it is in issue #182.

So, what I understand (other than the two other PRs) is this:

1. We only say `yes/no` and remove `skip`
2. When #182 is done, we'll have `yes/no/view` or `yes/no/print` (TBD)

So for this specific PR, we'll have:

```
The package xyz wants to run "chocolateyInstall.ps1"
Note: If you don't run this script, the installation will fail.
Do you want to run the script? (yes/no): n
```

I guess when #188 is done, we can remove that note.

Sounds good?

**Comment by ferventcoder** [comment]
Merged into stable at https://github.com/chocolatey/choco/commit/ccd3c65286d2abddad4c4e1a0a64429a5f25bc3d and will be released in 0.9.9.5.
